pkg/manager/member/pd_failover.go

```diff
@@ -163,50 +172,56 @@ func (f *pdFailover) tryToDeleteAFailureMember(tc *v1alpha1.TidbCluster) error {
 		return err
 	}
 	// invoke deleteMember api to delete a member from the pd cluster
-	err = controller.GetPDClient(f.deps.PDControl, tc).DeleteMemberByID(memberID)
-	if err != nil {
-		klog.Errorf("pd failover: failed to delete member: %d, %v", memberID, err)
+	if err := controller.GetPDClient(f.deps.PDControl, tc).DeleteMemberByID(memberID); err != nil {
+		klog.Errorf("tryToDeleteAFailureMember: failed to delete member %d, error: %v", memberID, err)
 		return err
 	}
-	klog.Infof("pd failover: delete member: %d successfully", memberID)
-	f.deps.Recorder.Eventf(tc, apiv1.EventTypeWarning, "PDMemberDeleted",
-		"%s(%d) deleted from cluster", failurePodName, memberID)
+	klog.Infof("tryToDeleteAFailureMember: delete member %s/%s(%d) successfully", ns, failurePodName, memberID)
+	f.deps.Recorder.Eventf(tc, apiv1.EventTypeWarning, "PDMemberDeleted", "failure member %s/%s(%d) deleted from PD cluster", ns, failurePodName, memberID)
 
 	// The order of old PVC deleting and the new Pod creating is not guaranteed by Kubernetes.
-	// If new Pod is created before old PVC deleted, new Pod will reuse old PVC.
-	// So we must try to delete the PVC and Pod of this PD peer over and over,
-	// and let StatefulSet create the new PD peer with the same ordinal, but don't use the tombstone PV
+	// If new Pod is created before old PVCs are deleted, the Statefulset will try to use the old PVCs and skip creating new PVCs.
+	// This could result in 2 possible cases:
+	// 1. If the old PVCs are first mounted successfully by the new Pod, the following pvc deletion will fail and return error.
+	//    We will try to delete the Pod and PVCs again in the next requeued run.
+	// 2. If the old PVCs are first deleted successfully here, the new Pods will try to mount non-existing PVCs, which will pend forever.
+	//    This is where OrphanPodsCleaner kicks in, which will delete the pending Pods in this situation.
+	//    Please refer to orphan_pods_cleaner.go for details.
 	pod, err := f.deps.PodLister.Pods(ns).Get(failurePodName)
 	if err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("tryToDeleteAFailureMember: failed to get pods %s for cluster %s/%s, error: %s", failurePodName, ns, tcName, err)
+		return fmt.Errorf("tryToDeleteAFailureMember: failed to get pod %s/%s for tc %s/%s, error: %s", ns, failurePodName, ns, tcName, err)
 	}
-
-	ordinal, err := util.GetOrdinalFromPodName(failurePodName)
-	if err != nil {
-		return err
+	if pod == nil {
+		klog.Infof("tryToDeleteAFailureMember: failure pod %s/%s not found, skip", ns, failurePodName)
+		return nil
```

Comment:
Why return here? What if the Pod was deleted but the PVCs failed to be deleted?

Reply:
I think we can first delete the PVCs then the pod to avoid the complexity? then we can return here safely.